### PR TITLE
Add stable messaging.kafka.offset attribute via SemconvStability (Fixes #15723)

### LINKED ISSUE #15723: Replace deprecated `messaging.kafka.message.offset` with `messaging.kafka.offset`
### Is your feature request related to a problem? Please describe.

Currently, Kafka instrumentation uses `messaging.kafka.message.offset` attribute which is deprecated in semantic conventions. It should be replaced with `messaging.kafka.offset`.
- https://opentelemetry.io/docs/specs/semconv/registry/attributes/messaging/#kafka-attributes

### Describe the solution you'd like

* Replace`messaging.kafka.message.offset` with `messaging.kafka.offset`

### Describe alternatives you've considered

_No response_

### Additional context

[Should messaging.kafka.message.offset be in the message namespace?](https://github.com/open-telemetry/semantic-conventions/issues/1156)

### Tip

<sub>[React](https://github.blog/news-insights/product-news/add-reactions-to-pull-requests-issues-and-comments/) with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it. Learn more [here](https://opentelemetry.io/community/end-user/issue-participation/).</sub>

### PR DESCRIPTION
**Pull request by Tusharika725**
Fixes #15723

This PR addresses the deprecated messaging string by deploying feature toggles for the Messaging Semantic Conventions and wiring them into the Kafka interceptors.

Changes:

Added emitStableMessagingSemconv() and emitOldMessagingSemconv() to SemconvStability.java.

Wired the stable messaging.kafka.offset attribute into KafkaProducerAttributesExtractor and KafkaConsumerAttributesExtractor while maintaining the old attribute for messaging/dup compatibility.

Note for Reviewers: I am contributing from a machine with strict memory constraints  and was unable to spin up the Testcontainers/Docker environment locally to verify KafkaClientDefaultTest without freezing my system. I am relying on the CI pipeline to catch the expected assertion failure so I can update the test accordingly. Thank you for your patience!

### CODE REVIEW

File: instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/build.gradle.kts [hunk 1]
```diff
@@ -66,6 +66,34 @@ tasks {
...
+    filter {
+      excludeTestsMatching("KafkaClientPropagationDisabledTest")
+      excludeTestsMatching("KafkaClientSuppressReceiveSpansTest")
+    }
+    jvmArgs("-Dotel.instrumentation.messaging.experimental.receive-telemetry.enabled=true")
+    jvmArgs("-Dotel.semconv-stability.opt-in=messaging")
+    systemProperty("metadataConfig", "otel.semconv-stability.opt-in=messaging")
+  }
+
+  val testBothSemconv by registering(Test::class) {
```
Comment (laurit) [inline instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/build.gradle.kts]: @trask wdyt should we test this with `messaging/dup` or is this overkill?

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SemconvStability.java [hunk 2]
```diff
@@ -54,8 +59,36 @@ public final class SemconvStability {
...
+    dbSystemNameMap.put("hanadb", "sap.hana");
+    dbSystemNameMap.put("informix", "ibm.informix");
+    dbSystemNameMap.put("ingres", "actian.ingres");
+    dbSystemNameMap.put("maxdb", "sap.maxdb");
+    dbSystemNameMap.put("mssql", "microsoft.sql_server");
+    dbSystemNameMap.put("netezza", "ibm.netezza");
+    dbSystemNameMap.put("oracle", "oracle.db");
+    dbSystemNameMap.put("redshift", "aws.redshift");
+    dbSystemNameMap.put("spanner", "gcp.spanner");
   }
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SemconvStability.java]: @Tusharika725 can you revert unrelated method ordering changes in this file? thanks

File: instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/build.gradle.kts (hunk 1 above)
Comment (trask) [reply instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/build.gradle.kts]: I'm ok removing it. We didn't test `http/dup` or `database/dup`. (we are only testing `rpc/dup` because of some additional edge cases in RPC)

**Comment by Tusharika725** [comment]
@trask @laurit I've removed the testBothSemconv block to keep it consistent  and I reverted the accidental IDE method reordering in SemconvStability.java. Let me know if anything else is needed!

**Review by laurit — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SemconvStability.java [hunk 3]
```diff
@@ -37,6 +37,9 @@ public final class SemconvStability {
   private static final boolean emitOldRpcSemconv;
   private static final boolean emitStableRpcSemconv;
 
+  private static final boolean emitOldMessagingSemconv;
+  private static final boolean emitStableMessagingSemconv;
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SemconvStability.java:43]: I think I'd prefer to put these breaking changes under v3 preview flag, since it's probably going to be a while for stable messaging, and I'd also like to get away from our pattern of adding "stable" flag but then introducing further breaking changes under the stable flag while we build out the stable telemetry

@laurit wdyt?

Comment (laurit) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SemconvStability.java:43]: It depends on how we are going to treat this. If we wish to do this change in v3 then v3 preview would be a good fit. If we treat this as part of the messaging semconv stabilization, which is not going to be ready for v3, then the semconv stability opt-in would be better. My understanding is that this is the first step for implementing the latest messaging semconv so using the opt-in flag seems fine to me.

Comment (trask) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SemconvStability.java:43]: > My understanding is that this is the first step for implementing the latest messaging semconv so using the opt-in flag seems fine to me.

ya, makes sense

I still don't love our existing practice of the `OTEL_SEMCONV_STABILITY_OPT_IN=database` flag not following our stability guidelines (it looks like a stable option, but it's not since it's essentially in-progress work)

messaging would be a good chance to switch, since the others will hopefully get wrapped up by 3.0 and this will hopefully be the only one to extend into 3.x

maybe we could use `OTEL_SEMCONV_STABILITY_PREVIEW=messaging`?

Comment (laurit) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SemconvStability.java:43]: changed to preview flag

**Review by trask — APPROVED** [review]
